src/Integration.Vsix/CFamily/ProcessRunner.cs

```diff
@@ -116,66 +120,87 @@ public bool Execute(ProcessRunnerArguments runnerArgs)
                     else
                     {
                         LogMessage(CFamilyStrings.MSG_ExecutionCancelled);
-                        return false;
+                        return;
                     }
                 }
 
-                var result = WaitForProcessToFinish(process, runnerArgs);
-                return result;
+                using (var timeoutTask = Task.Factory.StartNew(() => WaitForProcessToFinish(process, runnerArgs)))
+                {
+                    process.BeginErrorReadLine();
+
+                    // Warning: do not log the raw command line args as they
+                    // may contain sensitive data
+                    LogDebug(CFamilyStrings.MSG_ExecutingFile,
+                        runnerArgs.ExeName,
+                        runnerArgs.AsLogText(),
+                        runnerArgs.WorkingDirectory,
+                        runnerArgs.TimeoutInMilliseconds,
+                        process.Id);
+
+                    try
+                    {
+                        runnerArgs.HandleInputStream(process.StandardInput);
+                        runnerArgs.HandleOutputStream(process.StandardOutput);
+                    }
+                    catch (Exception) when (isRunningProcessCancelled)
```

Comment:
Without the catch, when an analysis was cancelled it would show an ugly stack trace to the user coming from Protocol.Read.

Reply:
`runnerArgs.HandleInputStream/HandleOutputStream` could be null. Using null-coalesce operators should be enough (unless we're actually relying on the streams really being processed).